### PR TITLE
C#: Use stubs for query test cases.

### PR DESCRIPTION
**Pull request by michaelnebel**
In this PR we re-factor the test cases in
- AlertSupression
- API Abuse

The only that was not convertible was "API Abuse/NoDisposeCallOnLocalIDisposable" as CIL extraction was required.

### CODE REVIEW

File: csharp/ql/test/query-tests/API Abuse/DisposeNotCalledOnException/options [hunk 1]
```diff
@@ -1,2 +1,2 @@
-semmle-extractor-options: /r:System.ComponentModel.Primitives.dll /r:${testdir}/../../../resources/assemblies/System.Data.dll /r:System.Data.Common.dll
```
Comment (hvitved) [inline csharp/ql/test/query-tests/API Abuse/DisposeNotCalledOnException/options:1]: Can we now remove some of these DLLs?

**Review by hvitved — APPROVED** [review]

File: csharp/ql/test/query-tests/API Abuse/DisposeNotCalledOnException/options (hunk 1 above)
Comment (michaelnebel) [reply csharp/ql/test/query-tests/API Abuse/DisposeNotCalledOnException/options:1]: Yes, that is actually possible. I will narrow the list to only what is strictly needed.

Comment (michaelnebel) [reply csharp/ql/test/query-tests/API Abuse/DisposeNotCalledOnException/options:1]: Ah - do you mean whether we can remove some of the DLLs from the repo?

Comment (hvitved) [reply csharp/ql/test/query-tests/API Abuse/DisposeNotCalledOnException/options:1]: Yes, that's what I meant.

**Pull request merged**